coremltools/converters/mil/frontend/torch/ops.py

```diff
@@ -1527,6 +1529,8 @@ def _avg_pool(context, node, inputs):
     x = inputs[0]
     kernel_sizes = inputs[1]
     strides = inputs[2]
+    if strides.op.op_type == "const"  and (not strides.val.any()):
```

Comment:
I was wondering why are not using the same condition here like we did for `max_pool2d`?

I mean, why `(not strides.val.any())` is being used whereas in `max_pool2d`, we are using `(not list(strides.val))`

Reply:
Good question. When you take out the .any() then the testcase fails (and the suggestion from Python was to add any()). Why there is this difference between max_pool2d and avg_pool2d behavior I am not sure but would appreciate insight. You can check yourself running the testcase with and without the any().

Reply:
`not list(tensor)` seems incorrect! here we are trying to infer if strides are set or not. And if set they are non-zero
e.g. `strides = [0, 0 ,0]` we would like to set kernel_size default. But do we see such case? @leovinus2001 which test is failing when you don't have this check? Torch should have stride input only when stride is valid.
IMO, we should not need to have a check on stride value if it's present.

Reply:
@bhushan23 
Does `not list(tensor)` check if strides is set or not? It would only be true is tensor is not set and false otherwise.

Can you gave a counter example where `not list(tensor)` seems incorrect?

Reply:
Thank you for your comments so far. To make sure we are on the same page, I use the test case https://github.com/apple/coremltools/files/5017284/testPool.txt
in issue #823 

That test case is testing 4 types of avg_pool and 4 types of max_pool. For avg_pool we test cases 2,3,6,7 of test case which are:
2)            y = F.avg_pool2d(x, 2)
3)            y = F.avg_pool2d(x, 2, stride=2)
6)        self.avg1 = nn.AvgPool2d(2,stride=2) 
7)        self.avg1b = nn.AvgPool2d(2)
        
With respect to code and patch in "def _avg_pool" at " strides = inputs[2]" the issue is that case 2 will have "strides.val = []" empty list while case 3,6,7 will have "strides.val = [2,2]. 

In other words, the patch is testing for an empty list there. We can do that with "not strides.val" if strides.val is a list, or  "not list(strides.val)" 
    if strides.op.op_type == "const"  and (not list(strides.val)):

PS: Same thing for max_poold2  with "y = F.max_pool2d(x, 2)" that will result in empty list for strides.

As I am retesting today again, I see that all 8 cases in the test case are now good via
    if strides.op.op_type == "const"  and (not list(strides.val)):
and we can forget about the .any()

hence, I propose to change 1 line from
   if strides.op.op_type == "const"  and (not strides.val.any()):
to
    if strides.op.op_type == "const"  and (not list(strides.val)):

Is that acceptable?

Reply:
Yes, looks good to me. Thanks : )

Reply:
Ok, I'll update the code then.